src/Microsoft.AspNetCore.Hosting/Internal/HostingLoggerExtensions.cs

```diff
@@ -18,7 +18,11 @@ internal static class HostingLoggerExtensions
 
         public static IDisposable RequestScope(this ILogger logger, HttpContext httpContext)
         {
-            return logger.BeginScope(new HostingLogScope(httpContext));
+            // to avoid allocation, return a null scope if the logger is not on at least to some degree.   
+            if (logger.IsEnabled(LogLevel.Critical))
```

Comment:
We did this in the past and it turned out to be completely broken. Do you recall why that was @rynowak ?